### PR TITLE
Feature/final analyses EE2

### PR DESCRIPTION
**Pull request by PerryShafran-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

This PR includes the final EE2 items from the EE2 spreadsheet, including:

- Adding WARNING in front of missing data statements and adding them to the output
- Putting in meaningful docblocks and comments
- Adding -p to mkdir comments to prevent "File exists" comments
- Changing cp to cpreq for the plot step

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Run the stats step, using qsub -v vhr=00 or your favorite cycle.
Run the plots step, setting COMIN to emc.vpppg.  

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

ASAP

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by ShelleyMelchior-NOAA** [comment]
I have cloned your fork and checked out branch `feature/analyses_moreEE2`

**Comment by ShelleyMelchior-NOAA** [comment]
The stats processing has completed. I ran it for vhr=04.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/analyses`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_*_grid2obs_stats.o107468*`

**Comment by PerryShafran-NOAA** [comment]
The stats output looks fine.

File: scripts/stats/analyses/exevs_analyses_rtma_grid2obs_stats.sh [hunk 1]
```diff
@@ -277,11 +301,14 @@ then
       cat mailmsg | mail -s "$subject" $maillist
     fi
    fi
+
+# Check for Guam urma file
```
Comment (ShelleyMelchior-NOAA) [inline scripts/stats/analyses/exevs_analyses_rtma_grid2obs_stats.sh]: Should this be "Guam rtma"?

File: scripts/stats/analyses/exevs_analyses_rtma_grid2obs_stats.sh [hunk 2]
```diff
@@ -292,6 +319,8 @@ then
    fi
 fi
 
+# Run METplus for urma vs obs
```
Comment (ShelleyMelchior-NOAA) [inline scripts/stats/analyses/exevs_analyses_rtma_grid2obs_stats.sh]: Should this be "rtma vs obs"?

File: scripts/stats/analyses/exevs_analyses_urma_grid2obs_stats.sh [hunk 3]
```diff
@@ -105,14 +119,15 @@ then
 	 fi
 	fi
 
-# check for CONUS rtma2p5 file
+# check for akurma urma file
```
Comment (ShelleyMelchior-NOAA) [inline scripts/stats/analyses/exevs_analyses_urma_grid2obs_stats.sh]: Should this be "Alaska urma"?

File: scripts/stats/analyses/exevs_analyses_urma_grid2obs_stats.sh [hunk 4]
```diff
@@ -153,14 +168,15 @@ then
 	 fi
 	fi
 
-	# check for CONUS rtma2p5 file
+	# check for hiurma urma file
```
Comment (ShelleyMelchior-NOAA) [inline scripts/stats/analyses/exevs_analyses_urma_grid2obs_stats.sh]: Should this be "Hawaii urma"?

File: scripts/stats/analyses/exevs_analyses_urma_grid2obs_stats.sh [hunk 5]
```diff
@@ -256,7 +279,7 @@ then
 	 cp $finalstat/evs.stats.${regionnest}${typtag}.${RUN}.${VERIF_CASE}.v${VDATE}.stat $COMOUTfinal
        fi
 else    
-       echo "NO RTMA OR OBS DATA, or not gather time yet, METplus gather job will not run"
+       echo "WARNING: NO RTMA OR OBS DATA, or not gather time yet, METplus gather job will not run"
```
Comment (ShelleyMelchior-NOAA) [inline scripts/stats/analyses/exevs_analyses_urma_grid2obs_stats.sh]: Should this be "URMA OR OBS DATA"?

File: scripts/stats/analyses/exevs_analyses_rtma_ru_grid2obs_stats.sh [hunk 6]
```diff
@@ -118,7 +136,7 @@ then
        fi
 
 else
-       echo "NO RTMA OR OBS DATA, or not gather time yet, METplus gather job will not run"
+       echo "WARNING: NO RTMA OR OBS DATA, or not gather time yet, METplus gather job will not run"
```
Comment (ShelleyMelchior-NOAA) [inline scripts/stats/analyses/exevs_analyses_rtma_ru_grid2obs_stats.sh]: Should this be "RTMA-RU OR OBS DATA"?

File: scripts/stats/analyses/exevs_analyses_rtma_grid2obs_stats.sh (hunk 1 above)
Comment (PerryShafran-NOAA) [reply scripts/stats/analyses/exevs_analyses_rtma_grid2obs_stats.sh]: Fixed.

File: scripts/stats/analyses/exevs_analyses_rtma_grid2obs_stats.sh (hunk 2 above)
Comment (PerryShafran-NOAA) [reply scripts/stats/analyses/exevs_analyses_rtma_grid2obs_stats.sh]: Fixed.

File: scripts/stats/analyses/exevs_analyses_urma_grid2obs_stats.sh (hunk 3 above)
Comment (PerryShafran-NOAA) [reply scripts/stats/analyses/exevs_analyses_urma_grid2obs_stats.sh]: Fixed.

**Comment by ShelleyMelchior-NOAA** [comment]
✔️ on the stats! Plots is running.

File: scripts/stats/analyses/exevs_analyses_rtma_ru_grid2obs_stats.sh (hunk 6 above)
Comment (PerryShafran-NOAA) [reply scripts/stats/analyses/exevs_analyses_rtma_ru_grid2obs_stats.sh]: Fixed.

File: scripts/stats/analyses/exevs_analyses_urma_grid2obs_stats.sh (hunk 5 above)
Comment (PerryShafran-NOAA) [reply scripts/stats/analyses/exevs_analyses_urma_grid2obs_stats.sh]: Fixed.

File: scripts/stats/analyses/exevs_analyses_urma_grid2obs_stats.sh (hunk 4 above)
Comment (PerryShafran-NOAA) [reply scripts/stats/analyses/exevs_analyses_urma_grid2obs_stats.sh]: Fixed.

**Comment by ShelleyMelchior-NOAA** [comment]
plots processing has completed.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/analyses/atmos.20231026`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_grid2obs_plots.o107470980`

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA @ShelleyMelchior-NOAA I have looked over this Final EE2 PR this morning and I think that it looks great! I approve this after all testing is completed. Thanks!

**Comment by PerryShafran-NOAA** [comment]
Plots looks good.  The tarball is the exact same size as the latest run.

**Comment by ShelleyMelchior-NOAA** [comment]
Awesome! I will do the formal code review and approve this PR for merge.

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
All changes look good. All tests yield expected results. This PR is approved for merge.

**Pull request merged**